moped-editor/src/utils/projectComponentHelpers.js

```diff
@@ -48,15 +48,13 @@ export const useLineRepresentable = mopedComponents => {
   const [lineRepresentable, setLineRepresentable] = useState([]);
   useEffect(() => {
     if (!mopedComponents) return;
-    let lineRepresentableNew = [
-      ...new Set(
-        mopedComponents.map(moped_component =>
-          moped_component?.line_representation
-            ? moped_component.component_name.toLowerCase()
-            : null
-        )
-      ),
-    ].filter(item => item);
+    const lineRepresentableNew = {}
+    mopedComponents.forEach(moped_component => {
+      const componentNameKey =
+        moped_component.component_name.toLowerCase().replaceAll(" ", "") +
+        (moped_component.component_subtype ?? "").toLowerCase().replaceAll(" ", "");
```

Comment:
This feels like a good place to create a helper that describes how this translation works to help ID components/subtypes, but I think that you are right to hold off on too much work here before potential refactoring. 👍